compiler/rustc_mir_transform/src/gvn.rs

```diff
@@ -644,12 +644,10 @@ impl<'body, 'tcx> VnState<'body, 'tcx> {
             {
                 if let Some(offset) = self.evaluated[idx].as_ref()
                     && let Ok(offset) = self.ecx.read_target_usize(offset)
+                    && let Some(min_length) = offset.checked_add(1)
                 {
-                    projection.to_mut()[i] = ProjectionElem::ConstantIndex {
-                        offset,
-                        min_length: offset + 1,
-                        from_end: false,
-                    };
+                    projection.to_mut()[i] =
+                        ProjectionElem::ConstantIndex { offset, min_length, from_end: false };
```

Comment:
Do you know why `ConstantIndex` stores `min_length` instead of, e.g., the "max index" of the last value?

Reply:
Well I guess it doesn't make sense to index at `usize::MAX` anyways.

Reply:
`min_length` comes from slice match. I supposed it's easier to compute from the length of the pattern.